### DISCUSSION TITLE
Error with deploy version WPF app

### DISCUSSION THREAD

**Discussion by recetarioMDWiki** [cefsharp/CefSharp#4047] [General]
Hi everyone!
I have an WPF application (with MahApps) and I added the CefSharp nuget packet. All works fine in debug/release mode in developer machine, but when I publish the app, throw me this exception:

>Error: Invoking the constructor of type 'CefSharp.Wpf.ChromiumWebBrowser' matching the specified binding constraints caused an exception. in System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
in System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   in System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parint, Boolean closeStream)
   in System.Windows.Application.LoadComponint(Object componint, Uri resourceLocator)
   in GestionMedicamintos.Variaciones.InitializeComponint()
   in GestionMedicamintos.Variaciones..ctor()
   in GestionMedicamintos.Continedor_Principal_Var.CargaSolicitud()
   in GestionMedicamintos.Continedor_Principal_Var.CargaSolicitud()
   in GestionMedicamintos.Continedor_Principal_Var.VB$StateMachine_24_Continedor_Principal_Var_Loaded.Movinext()

It's .NET Framework 4.5.2.
It's AnyCPU.
It's deploying with ClickOnce.
I have reviewed the readme.txt after Nuget installation.
I have reviewed that all necessary files are copied at bin/Release directory.

This is the xalm:
![image](https://user-images.githubusercontent.com/50943477/160362875-fd7a7c73-f4bf-4b8c-9e5c-faf4742b373a.png)
![image](https://user-images.githubusercontent.com/50943477/160363121-09f29f38-37f5-46e8-be78-168c1a34d104.png)

In the code (.vb) I only use the WebBrowser to load an html string:
![image](https://user-images.githubusercontent.com/50943477/160363351-9bf6adf3-434a-4359-95da-e71e189681af.png)

If anyone has a clue as to what I'm doing wrong or what I'm missing it would be a great help. 
Thanks to all!

**Comment by amaitland** [comment]
> It's AnyCPU. It's deploying with ClickOnce.

AnyCPU doesn't currently work when deployed via click once. You'll need to target x86 or x64. Extra steps maybe required depending on if you are using packages.config or PackageReference. There are multiple discussions that cover the steps.

**Reply by recetarioMDWiki** [reply]
Hi Alex, thanks for your reply.

In this moment I can not guarantee that all machines with I work are x86 or x64, so for now, I do not will use CefSharp. But I will be attentive to the following versions.
Thanks for all your work and the community efforts.

Regards!

**Reply by amaitland** [reply]
> In this moment I can not guarantee that all machines with I work are x86 or x64

x86 will run everywhere. Make your application is large address aware see https://github.com/cefsharp/CefSharp/wiki/General-Usage#win32-out-of-memory

**Reply by amaitland** [reply]
I've created https://github.com/cefsharp/CefSharp/discussions/4056 which has details on how to get `AnyCPU` working.
